syncapi/storage/postgres/syncserver.go

```diff
@@ -109,42 +110,85 @@ func (d *SyncServerDatasource) Events(ctx context.Context, eventIDs []string) ([
 
 	// We don't include a device here as we only include transaction IDs in
 	// incremental syncs.
-	return streamEventsToEvents(nil, streamEvents), nil
+	return d.StreamEventsToEvents(nil, streamEvents), nil
 }
 
 // WriteEvent into the database. It is not safe to call this function from multiple goroutines, as it would create races
 // when generating the sync stream position for this event. Returns the sync stream position for the inserted event.
 // Returns an error if there was a problem inserting this event.
+// nolint: gocyclo
```

Comment:
I'd rather we factor out some bits of that function than skip the check, otherwise we'll easily come up with unmaintainable functions that do way too much stuff by themselves.